### ISSUE TITLE
week #10

### ISSUE THREAD

**Issue by joonsun-baek** [ml-mate/coursera-machine-leaning#17]
### Batch Gradient Descent
* 기존에 알고있는 Gradient Descent
* 식을 정리하자면 아래와 같음
  <img width="921" alt="스크린샷 2019-09-26 오전 10 50 32" src="https://user-images.githubusercontent.com/18119192/65652521-03a8ab80-e04d-11e9-8956-474ca203ef55.png">

###  Stochastic Gradient Descent
* SGD와 BGD의 차이는 theta를 구하는 식의 차이를 둔다고 짧게 설명할 수 있는데,
  간단히 말해 `1/(2*m) * SUM` 을 제거하여, 모든 m을 돌지 않고 각 개별의 sample `i`에 대해 `x(i) - y(i)`만으로 theta의 업데이트를 하도록 하는 방법
  * BGD의 cost function
    <img width="286" alt="스크린샷 2019-09-26 오전 10 55 39" src="https://user-images.githubusercontent.com/18119192/65652373-7d8c6500-e04c-11e9-9914-39980fee565b.png">

  * SGD의 cost function
    <img width="161" alt="스크린샷 2019-09-26 오전 10 55 44" src="https://user-images.githubusercontent.com/18119192/65652293-369e6f80-e04c-11e9-919f-ab6c7b8b64fd.png">

* 각 `i`마다 theta가 변경되기 때문에, `J`를 계산하기 전에, train sample을 random-shuffle 을 먼저 해야함. 따라서 각 sample `i`의 계산마다 theta가 잘 학습되게 되어야 하기 때문

* 식을 말하자면 아래와 같음
  <img width="919" alt="스크린샷 2019-09-26 오전 10 50 38" src="https://user-images.githubusercontent.com/18119192/65652070-8597d500-e04b-11e9-993f-52dd20ec413a.png">

### Mini-batch Gradient Descent
* BGD와 SGD의 중간처럼 활용하는 방법
* batch size만큼씩 theta를 업데이트 시키는 연산을 수행
* 간단히 말해, batch-size가 10이라면, 
  <img width="219" alt="스크린샷 2019-09-26 오전 11 03 51" src="https://user-images.githubusercontent.com/18119192/65652640-5f733480-e04d-11e9-988b-5d6fb18fc4f3.png">
  이렇게 수행
* 식을 정리하자면 아래와 같음
  <img width="813" alt="스크린샷 2019-09-26 오전 11 02 12" src="https://user-images.githubusercontent.com/18119192/65652562-28048800-e04d-11e9-8f85-a27c65235f41.png">

## Online Learning
* 실시간으로 인입되는 샘플을 theta에 반영시켜, 모델을 실시간으로 맞춰가는 기법
  * 실제 theta의 업데이트는 SGD와 비슷하게 수행.

  <img width="762" alt="스크린샷 2019-09-26 오전 11 06 06" src="https://user-images.githubusercontent.com/18119192/65652718-a95c1a80-e04d-11e9-83bf-35b4dc0bd2fa.png">

## MR
* 요건 MapReduce에 대한 내용, 패스
